ethers/wallet.nim

```diff
@@ -29,10 +30,16 @@ proc new*(_: type Wallet, privateKey: PrivateKey, provider: Provider): Wallet =
   let wallet = Wallet.new(privateKey)
   wallet.provider = some provider
   wallet
-proc new*(_: type Wallet, privateKey: string): Wallet =
-  Wallet.new(PrivateKey.fromHex(privateKey).value)
-proc new*(_: type Wallet, privateKey: string, provider: Provider): Wallet =
-  Wallet.new(PrivateKey.fromHex(privateKey).value, provider)
+proc new*(_: type Wallet, privateKey: string): ?Wallet =
+  if key =? PrivateKey.fromHex(privateKey):
+    some Wallet.new(key)
+  else:
+    none Wallet
+proc new*(_: type Wallet, privateKey: string, provider: Provider): ?Wallet =
+  if key =? PrivateKey.fromHex(privateKey):
+    some Wallet.new(key, provider)
+  else:
+    none Wallet
```

Comment:
Would it make more sense to return a `?!Wallet` instead? The `fromHex` error could be mapped. I think I find it odd that a constructor returns an option, which lacks context as to why, and the caller is left with a `none Wallet` and no way to know it was because of a bad private key.

Reply:
Good idea, thanks! I've changed it to return a Result instead of an Option.